Sources/SwiftSentry/Protcols/SentryValueSerializable.swift

```diff
@@ -0,0 +1,11 @@
+// SPDX-License-Identifier: BSD-3-Clause
+import sentry
+
+import Foundation
+
+// A protocol that allows for Sentry objects
+// to easily produce their serialized value.
+protocol SentryValueSerializable {
```

Comment:
The Sentry Cocoa SDK relies pretty heavily on some serializable protocols, and I like the idea of have a non-public one to make it easier for pulling out the required C structures, but I don't think it's a thing that anyone consuming this SDK should have to think about.

Reply:
Totally cool to have internal protocols. Would you mind explicitly making this private?

Reply:
If this is marked as `private` will I still be able to call it in tests? I thought that something has to be `internal` to get prompted to `public` when you use `@testable`